src/lib/Runner.ts

```diff
@@ -128,6 +129,16 @@ export class Runner {
       return landRequest.setStatus('fail', 'Unable to land due to failed land checks');
     }
 
+    if (
+      await SpeculationEngine.reorderRequest(
+        runningTargetingSameBranch,
+        await this.getQueue(['queued']),
```

Comment:
Just realised something. Should we also be filtering the queue to the same target branch?

Reply:
good catch. Yes, we should be filtering the queue to the same target branch.